src/nacl/signing.py

```diff
@@ -29,26 +31,45 @@ class SignedMessage(bytes):
     """
 
     @classmethod
-    def _from_parts(cls, signature, message, combined):
+    def _from_parts(cls, signature, message, combined, encoder):
         obj = cls(combined)
+        obj._encoder = encoder
         obj._signature = signature
         obj._message = message
         return obj
 
     @property
-    def signature(self):
+    def raw_signature(self):
         """
         The signature contained within the :class:`SignedMessage`.
         """
         return self._signature
 
     @property
-    def message(self):
+    def raw_message(self):
         """
         The message contained within the :class:`SignedMessage`.
         """
         return self._message
 
+    @property
+    def signature(self):
+        """
+        The signature contained within the :class:`SignedMessage`,
+        encoded as requested on signature generation
+        """
+        warn("Deprecated attribute")
```

Comment:
This is an absolutely terrible warning message. It doesn't tell the user (or developer) _which_ attribute is deprecated (is it `signature`? `message`? `raw_signature`? `raw_message`? `__get__`? `__mro__`? `__getattr__`?), and it doesn't hint at what the correct, non-deprecated replacements are.

A much better warning message would be "Use of the signature property is deprecated. Please update your code to use the raw_signature attribute instead, and decode it properly."

Same for the `message` property below.

Reply:
@blag while I appreciate your interest in pynacl, before investing more time in this change, we must agree it goes in the right direction for our users.

The intent is to just deprecate the encoded parts, while keeping the raw parts for advanced usage, therefore I only added warnings on access to .message and .signature.

If @pyca/pynacl-core find some merit in this proposal, I'll be grateful for better message suggestion.